libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/target/RefreshSyncDownTarget.java

```diff
@@ -57,49 +57,33 @@ public class RefreshSyncDownTarget extends SyncDownTarget {
     public static final String FIELDLIST = "fieldlist";
     public static final String SOBJECT_TYPE = "sobjectType";
     public static final String SOUP_NAME = "soupName";
-    public static final String COUNT_IDS_PER_SOQL = "coundIdsPerSoql";
+    public static final String COUNT_IDS_PER_SOQL = "countIdsPerSoql";
     private List<String> fieldlist;
     private String objectType;
     private String soupName;
     private int countIdsPerSoql;
-    private static final int defaultCountIdsPerSoql = 500;
-    private static final int MAX_COUNT_IDS_PER_SOQL = 2000;
+    private static final int MAX_COUNT_IDS_PER_SOQL = 500;
+    // NB: SOQL query length limit is 100k but not over REST
+    //     Too many ids will cause a "414 - URI Too Long" error
 
     // NB: For each sync run - a fresh sync down target is created (by deserializing it from smartstore)
     // The following members are specific to a run
     // page will change during a run as we call start/continueFetch
     private boolean isResync = false;
     private int page = 0;
 
-    /**
-     * Return number of ids to pack in a single SOQL call
-     */
-    public int getCountIdsPerSoql() {
```

Comment:
These methods were only there to support tests. Replacing them with a package level constructor where one can specify the count ids per soql.